pkg/controller/controllerutil/controllerutil.go

```diff
@@ -281,3 +281,24 @@ func RemoveFinalizerWithError(o runtime.Object, finalizer string) error {
 	RemoveFinalizer(m, finalizer)
 	return nil
 }
+
+// ContainsFinalizer checks a metav1 object that the provided finalizer is present.
+func ContainsFinalizer(o metav1.Object, finalizer string) bool {
+	f := o.GetFinalizers()
+	for _, e := range f {
+		if e == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsFinalizerWithError checks a metav1 object that the provided finalizer is present.
+// It returns an error if the provided object cannot provide an accessor.
+func ContainsFinalizerWithError(o runtime.Object, finalizer string) (bool, error) {
```

Comment:
Why is this needed? All concrete api types, be they built-in or CRDs, have am emedded `metav1.ObjectMeta` and thus fulfill the `metav1.Object` interface. If you really end up writing generic code that acts on a `runtime.Object` you can just assert it as `metav1.Object` prior to trying to find out if it has the given finalizer.

The only type of `runtime.Object` where this will not work is list types, but that makes sense as they do not implement the `metav1.Object` interface. IMHO it would be better to tell ppl at compile time that this isn't possible for Lists rather than at runtime when `meta.Accessor` errors.

Reply:
I asked for this to be added to align with the other existing `*WithError` functions in this package.

I agree, though, that compile time type checking is better than runtime type assertions. #898 may help with this.

Reply:
#898 merged so maybe we should remove `ContainsFinalizerWithError` again and update `ContainsFinalizer` to take an `controllerutil.Object` as first agument?

Reply:
@alvaroaleman , thanks, I will do. Do we need also delete other *WithError methods and replace with controllerutil.Object methods which are not handling errors?

Reply:
For now, please leave the existing code as it is. Removing the `WithError` funcs that we already have is a breaking change, so we should do it in a dedicated PR to make sure it gets its own release note and only gets merged when we are ready to take breaking changes.

Reply:
Fixed

Reply:
Thank you!
/lgtm